driver/flambda_backend_flags.ml

```diff
@@ -24,6 +26,7 @@ module Flambda2 = struct
     let backend_cse_at_toplevel = false
     let cse_depth = 2
     let treat_invalid_code_as_unreachable = false
+    let function_result_types = Functors_only
```

Comment:
```suggestion
    let function_result_types = Never
```
(You said that you were expecting to only set it at `-O3`)

Reply:
Fixed